### DISCUSSION TITLE
Pressing the button

### DISCUSSION THREAD

**Discussion by Mikhail666999** [gilzoide/godot-lua-pluginscript#38] [Q&A] [answered]
Hello, it's me again) This time I have a problem in how to make a reaction to a click?
in gdscript:
`if Input.is_action_pressed("ui_accept"):
 #body
pass`

and in lua how?

**Comment by gilzoide** [comment, marked as answer]
Like described in [From GDScript to Lua](https://github.com/gilzoide/godot-lua-pluginscript/blob/main/extras/docs/from-gdscript-to-lua.md), Godot's singleton objects like `Input` are all readily available to Lua. The only change is that you need to use Lua's method notation with a `:` instead of `.`:

```lua
if Input:is_action_pressed("ui_accept") then
    -- do something
end
```

If in doubt, you can always use the Lua REPL bundled in the plugin to test out these syntax things.
It's located at the bottom tab of the editor.

https://user-images.githubusercontent.com/4603360/211172726-14f69534-2c3e-4421-8bbf-f18c97f4ff46.mov

**Reply by Mikhail666999** [reply]
Спасибо, я не думал, что в `.` и `:` есть такая разница
Thank you, I didn't think that in `.` and `:` there is such a difference)

**Reply by gilzoide** [reply]
Yeah, it does. This is something that Lua has different from most languages and may feel somewhat alien.
When you use only `.`, you only get the method object. That's why in the video you see a `table: 0x...` log: it's the [MethodBindingByName](https://github.com/gilzoide/godot-lua-pluginscript/blob/main/src/godot_class.lua#L209-L233) table that will call the method by name in the passed object. (yeah, we should add a `__tostring` to it to get a nicer log like `MethodBindByName<method_name>`)

So, for using `.`, the way would be to pass `Input` to the method explicitly:
```lua
Input.is_action_pressed(Input, "ui_accept")
                     -- ^ here!
```

This is super annoying, so Lua devs added this syntax sugar where you use `:` to call the method passing the object as its first argument at the same time:
```lua
Input:is_action_pressed("ui_accept")
```
Both code snippets do exactly the same, as far as Lua is concerned. More info here: https://www.lua.org/manual/5.1/manual.html#2.5.8
